website/docs/docs/build/data-tests.md

```diff
@@ -70,6 +70,30 @@ The name of this test is the name of the file: `assert_total_payment_amount_is_p
 
 Singular data tests are easy to write—so easy that you may find yourself writing the same basic structure over and over, only changing the name of a column or model. By that point, the test isn't so singular! In that case, we recommend...
 
+<VersionBlock firstVersion="1.9">
+
+### Document singular tests
+
+Previously, documenting a data test such as a singular data test would result in a parsing error. However, in [Versionless](/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can set descriptions in your YAML files to document singular data tests.
+
+<File name='data_tests.yml'>
+
+```yml
+
+# top-level
+data_tests:
+  - name: my_super_cool_singular_test
+    description: The order_id is unique for every row in the orders model
+    config:
+      error_if: ">10"
+  - name: my_super_cool_singular_test2
+    description: '{{ doc("not_null_test") }}'
```

Comment:
@aranke Can you confirm that docs blocks will work for documenting singular tests as well?

Reply:
## TLDR
We have two issues open for adding `description` to data tests:
- https://github.com/dbt-labs/dbt-core/issues/9005#issuecomment-1834047344
- https://github.com/dbt-labs/dbt-core/issues/2578

This PR _only_ allows the first and not the second.

## Details

There are two types of data tests:
1. Generic
1. Singular

### Descriptions for generic data tests

Generic data tests have one macro _definition_ and then zero to many _instances_ where they are used. So that's two different places we can imagine adding a description for generic data tests. My understanding is the first is supported by this PR but the second is **not**.

Since the _definition_ is just a macro, the definition of a custom generic data test can already have a [`description`](https://docs.getdbt.com/reference/resource-properties/description) like this:

```yaml
macros:
  - name: test_CUSTOM_GENERIC_TEST_NAME
    description: markdown_string
    arguments:
      - name: argument_name
        description: markdown_string
```

However, I don't believe the _instances_ of generic data tests can be documented until https://github.com/dbt-labs/dbt-core/issues/2578 is implemented.

### Descriptions for singular data tests

Added in https://github.com/dbt-labs/dbt-core/pull/10792 to resolve https://github.com/dbt-labs/dbt-core/issues/9005!

I will test out this example to make sure it works 👍

Reply:
✅ Works